app/admin/address/route.js

```diff
@@ -8,24 +8,26 @@ export default AbstractEditRoute.extend({
   newTitle: t('admin.address.newTitle'),
   editTitle: t('admin.address.editTitle'),
   model() {
-    return new Ember.RSVP.Promise((resolve, reject) => {
-      this.get('store').find('option', 'address_options').then((addressOptions) => {
-        resolve(addressOptions);
-      }, (err) => {
-        if (err instanceof UnauthorizedError) {
-          reject(err);
-        } else {
-          let store = this.get('store');
-          let newConfig = store.push(store.normalize('option', {
-            id: 'address_options',
-            value: {
-              address1Label: this.get('i18n').t('admin.address.addressLabel'),
-              address1Include: true
-            }
-          }));
-          resolve(newConfig);
-        }
-      });
+    return this.get('fetchAddressOptions').perform();
+  },
+  fetchAddressOptions: task(function* () {
+    let store = this.get('store');
+    try {
+      return yield store.find('option', 'address_options');
+    } catch(err) {
+      if (err instanceof UnauthorizedError) {
+        throw err;
+      }
+      return store.push(this.get('defaultAddressOption'));
+    }
+  }).keepLatest().cancelOn('deactivate'),
+  defaultAddressOption: function() {
+    return this.get('store').normalize('option', {
+      id: 'address_options',
+      value: {
+        address1Label: this.get('i18n').t('admin.address.addressLabel'),
+        address1Include: true
+      }
     });
-  }
+  }.property()
```

Comment:
Absolutely! I prefer DockYard's style better. I only kept this because I thought it was the style choice based on looking at the code.